components/meter/meter-linear.js

```diff
@@ -32,23 +49,61 @@ class MeterLinear extends LitElement  {
 				position: absolute;
 				top: 0;
 			}
+
+			.d2l-meter-linear-text {
+				color: var(--d2l-color-ferrite);
+				display: flex;
+				flex-direction: row;
+				justify-content:space-between;
+				margin: 0 6px;
+			}
+
+			.d2l-meter-linear-secondary {
+				align-self: flex-end;
+			}
+
 		`];
 	}
 
 	constructor() {
 		super();
-		this.value = 0;
 		this.max = 100;
+		this.percent = false;
+		this.singleLine = false;
+		this.value = 0;
 	}
 
 	render() {
 		const progressBarWidth = this.value / this.max * 100;
+		const primary = this.percent ? `${Math.floor(this.value / this.max * 100)}%` : `${this.value}/${this.max}`;
+		const secondaryText = this._formatContext(this.context, this.value, this.max);
+		const secondaryTextElement = secondaryText ? html`<div class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
+		const classSingleLine = this.singleLine ? 'd2l-meter-linear-flex' : 'd2l-meter-linear-block';
 		return html `
-			<div class="d2l-meter-linear-full-bar">
-				<div class="d2l-meter-linear-inner-bar" style="width:${progressBarWidth}%;"></div>
+			<div class="${classSingleLine}" role="img" aria-label="${secondaryText}: ${primary}; progress indicator">
```

Comment:
The progress role does not fit this use case :(

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role